plugins/googleanalytics/src/org/pentaho/di/trans/steps/googleanalytics/GaInputStep.java

```diff
@@ -0,0 +1,266 @@
+          }
+        } else {
+          // Assume it's a Dimension or Metric, we've covered the rest of the cases above.
+          dataObject = entry.get( j++ );
+        }
+        outputRow[i] = data.outputRowMeta.getValueMeta( i ).convertData( data.conversionMeta[i], dataObject );
+      }
+
+      // copy row to possible alternate rowset(s)
+      putRow( data.outputRowMeta, outputRow );
+
+      // Some basic logging
+      if ( checkFeedback( getLinesWritten() ) ) {
+        if ( log.isBasic() ) {
+          logBasic( "Linenr " + getLinesWritten() );
+        }
+      }
+      return true;
+
+    } else {
+      setOutputDone();
+      return false;
+    }
+  }
+
+  protected Analytics.Data.Ga.Get getQuery( Analytics analytics ) {
+
+    Analytics.Data dataApi = analytics.data();
+    Analytics.Data.Ga.Get query;
+
+    try {
+      query = dataApi.ga().get(
+        meta.isUseCustomTableId() ? environmentSubstitute( meta.getGaCustomTableId() ) : meta.getGaProfileTableId(), //ids
+        environmentSubstitute( meta.getStartDate() ), // start date
+        environmentSubstitute( meta.getEndDate() ), // end date
+        environmentSubstitute( meta.getMetrics() ) // metrics
+      );
+      query.setDimensions( environmentSubstitute( meta.getDimensions() ) );
+
+
+      if ( meta.isUseSegment() ) {
+        if ( meta.isUseCustomSegment() ) {
+          query.setSegment( environmentSubstitute( meta.getCustomSegment() ) );
+        } else {
+          query.setSegment( meta.getSegmentId() );
+        }
+      }
+
+      if ( !Const.isEmpty( meta.getFilters() ) ) {
+        query.setFilters( environmentSubstitute( meta.getFilters() ) );
+      }
+      if ( !Const.isEmpty( meta.getSort() ) ) {
+        query.setSort( environmentSubstitute( meta.getSort() ) );
+      }
+
+      return query;
+    } catch ( IOException ioe ) {
+      return null;
+    }
+
+  }
+
+  public boolean init( StepMetaInterface smi, StepDataInterface sdi ) {
+    meta = (GaInputStepMeta) smi;
+    data = (GaInputStepData) sdi;
+
+    if ( !super.init( smi, sdi ) ) {
+      return false;
+    }
+
+    // Look for deprecated field types and log error(s) for them
+    String[] types = meta.getFeedFieldType();
+    if ( types != null ) {
+      for ( String type : types ) {
+        if ( GaInputStepMeta.DEPRECATED_FIELD_TYPE_CONFIDENCE_INTERVAL.equals( type ) ) {
+          logError( BaseMessages.getString( PKG, "GoogleAnalytics.Warn.FieldTypeNotSupported",
+            GaInputStepMeta.DEPRECATED_FIELD_TYPE_CONFIDENCE_INTERVAL ) );
+        }
+      }
+    }
+
+    try {
+      // Create an Analytics object, and fetch what we can for later (account name, e.g.)
+      analytics = GoogleAnalyticsApiFacade.createFor( environmentSubstitute( meta.getGaAppName() ), meta )
+        .getAnalytics();
+      accountName = analytics.management().accounts().list().execute().getItems().iterator().next().getName();
```

Comment:
If any return null, we're screwed, so the catch block will get the NPE and return false